src/spox/_public.py

```diff
@@ -194,16 +202,45 @@ def inline(model: onnx.ModelProto) -> _InlineCall:
     unless they are used as a default argument value. In this case
     they are also built as initializers.
 
+    Symbolic dimensions in input and output shapes are stripped from
+    the model and ignored, to avoid handling them inconsistently.
+
     Build behaviour should be treated as an implementation detail and
     may change.
 
+    Currently, inlining models with subgraphs or functions is not supported
+    as it cannot be performed in most cases due to lack of upstream support.
     """
     in_names = [i.name for i in model.graph.input]
     in_defaults = {i.name: i for i in model.graph.initializer}
     out_names = [o.name for o in model.graph.output]
     _defaults_msg = f" (defaults {list(in_defaults.keys())})"
     _signature_msg = f"signature {in_names}{_defaults_msg} -> {out_names}"
 
+    model = _copy_model(model)
+    # FIXME: Renaming does not work on subgraphs as of ONNX 1.13/1.14.
+    for node in model.graph.node:
+        for attr in node.attribute:
+            if attr.HasField("g") or attr.graphs:
+                raise ValueError(
+                    "Inlining models with subgraphs is not supported due to "
+                    "lack of upstream support for renaming values in subgraphs."
+                )
+    # FIXME: Support for functions is a bit involved, as it interacts with build.
+    if model.functions:
+        raise ValueError(
+            "Inlining models with functions is not supported. "
```

Comment:
I think this could be made to work already with prefixing the function names, and upstream renaming might actually support this already. But we don't have a use case so I didn't get around to it.
Unless you mean something like this? https://github.com/onnx/onnx/pull/5105
In terms of upstream support renaming within subgraphs is what could really use upstream support, as well as arbitrary renaming instead of just prefixing.